fix:[CDS-89531]: Update rollback deployment doc with a limitation (#6483)
harness/developer-hub: docs/continuous-delivery/manage-deployments/rollback-deployments.md

* Rollback steps can use expressions that refer to steps executed during the previous pipeline execution. The expressions are automatically resolved with values from the original execution.
* The stages should roll back in reverse order when  deployment rollback is triggered.
* Rolling back to a previous post-production deployment is only possible for executions that occurred within the past 30 days.

> * If the failure strategy marks an execution as successful and ignored the failures, it doesn't mean you can roll back that execution.

Reviewer: ```suggestion
* You cannot roll back an execution if was marked successful by the ignore failures failure strategy.
```

Reviewer: Your statement sounds more casual. Please see if the suggestion is technically accurate.

Author: Hey @roshnisarangadharan we need to mention that if we use action mark as success.

Author: Yes @roshnisarangadharan look good
